### PR TITLE
Added __attribute__((visibility("default"))) to methods in TWString.h

### LINKED ISSUE #2013: [Android]  change set(CMAKE_CXX_VISIBILITY_PRESET hidden) to set(CMAKE_CXX_VISIBILITY_PRESET default)
I am not so familiar with cmake but I am wondering what implications changing:
`set(CMAKE_CXX_VISIBILITY_PRESET hidden)` to ` set(CMAKE_CXX_VISIBILITY_PRESET default)`
would have? (in CMakeLists.txt)

The setting is causing dart ffi to throw lookup errors on android (when using TrustWalletCore to develop flutter applications).

By changing the setting it would allow the flutter community to directly pull the library from this repo instead of building and hosting a custom library for flutter.

### PR DESCRIPTION
**Pull request by hnord-vdx**
## Description

All methods in `TWString.h` have visibility set to default using ` __attribute__((visibility("default")))`. This allows dart ffi (for flutter applications) to find the methods. When the attribute is not set dart ffi throws lookup errors. 

Fixes #2013 

## Testing instructions

No new tests needed. Run normal android and iOS tests

## Types of changes

* Bug fix (non-breaking change which fixes an issue) 

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

File: include/TrustWalletCore/TWBase.h [hunk 1]
```diff
@@ -14,6 +14,9 @@
 #endif
 #endif
 
+// Marker to make methods visible to dart ffi
```
Comment (hewigovens) [inline include/TrustWalletCore/TWBase.h:17]:
```suggestion
// Marker for default  visibility
```

**Review by hewigovens — COMMENTED** [review]
I think `TWData.h` also need those markers?

**Review by catenocrypt — APPROVED** [review]

File: include/TrustWalletCore/TWBase.h (hunk 1 above)
Comment (hnord-vdx) [reply include/TrustWalletCore/TWBase.h:17]: I don't know which functions use methods in `TWData.h` but I was able to generate wallets / addresses, sign transactions, verify addresses/mnemonic for different blockchains without any lookup errors. 

Do you know a function I can test on flutter android that uses methods in `TWData.h` to make sure?

Comment (hewigovens) [reply include/TrustWalletCore/TWBase.h:17]: just add same marker to `TWData.h`, `TWString` / `TWData` is just `const void`, others interfaces are good

File: include/TrustWalletCore/TWBase.h [hunk 2]
```diff
@@ -14,6 +14,9 @@
 #endif
 #endif
 
+// Marker for default  visibility
```
Comment (hewigovens) [inline include/TrustWalletCore/TWBase.h:17]:
```suggestion
// Marker for default visibility
```

File: include/TrustWalletCore/TWData.h [hunk 3]
```diff
@@ -10,63 +10,63 @@
 
 TW_EXTERN_C_BEGIN
 
-typedef const void TWString;
+typedef const void TWString TW_VISIBILITY_DEFAULT;
```
Comment (hewigovens) [inline include/TrustWalletCore/TWData.h:13]:
```suggestion
typedef const void TWString;
```

File: include/TrustWalletCore/TWData.h [hunk 4]
```diff
@@ -10,63 +10,63 @@
...
 
 /// Defines a resizable block of data.
 ///
 /// The implementantion of these methods should be language-specific to minimize translation overhead. For instance it
 /// should be a `jbyteArray` for Java and an `NSData` for Swift.
-typedef const void TWData;
+typedef const void TWData TW_VISIBILITY_DEFAULT;
```
Comment (hewigovens) [inline include/TrustWalletCore/TWData.h:19]:
```suggestion
typedef const void TWData;
```

**Review by hewigovens — APPROVED** [review]

**Pull request merged**